vortex-file/src/read/handle.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::BTreeSet;
+use std::sync::{Arc, RwLock};
+
+use futures::stream;
+use itertools::Itertools;
+use vortex_dtype::DType;
+use vortex_error::{VortexResult, VortexUnwrap as _};
+use vortex_io::{IoDispatcher, VortexReadAt};
+
+use super::splits::SplitsAccumulator;
+use super::{LayoutMessageCache, LayoutReader, LazyDType, RowMask, VortexReadArrayStream};
+use crate::read::buffered::{BufferedLayoutReader, ReadArray};
+use crate::read::splits::ReadRowMask;
+
+#[derive(Clone)]
+pub struct VortexReadHandle<R> {
+    input: R,
+    dtype: Arc<LazyDType>,
+    row_count: u64,
+    splits: Vec<(usize, usize)>,
+    layout_reader: Arc<dyn LayoutReader>,
+    filter_reader: Option<Arc<dyn LayoutReader>>,
+    messages_cache: Arc<RwLock<LayoutMessageCache>>,
+    row_mask: Option<RowMask>,
```

Comment:
is it necessary to provide a handle-level row-mask? This sort of feels like a parameter you pass when you build a scan from the handle

Reply:
The flow I imagine is something like:
1. Read some external index to create initial `RowMask`
2. Pass that row to the the builder to create a handle.
3. Reuse the initial mask to create concurrent reads on subranges of the file